pkg/csi/service/wcpguest/controller.go

```diff
@@ -420,6 +456,137 @@ func (c *controller) ControllerPublishVolume(ctx context.Context, req *csi.Contr
 	return resp, nil
 }
 
+// controllerPublishForFileVolume is a helper mthod for handling ControllerPublishVolume request for File volumes
+func controllerPublishForFileVolume(ctx context.Context, req *csi.ControllerPublishVolumeRequest, c *controller) (
+	*csi.ControllerPublishVolumeResponse, error) {
+	log := logger.GetLogger(ctx)
+	// Build the CnsFileAccessConfig instance name and namespace
+	cnsFileAccessConfigInstance := &cnsfileaccessconfigv1alpha1.CnsFileAccessConfig{}
+	cnsFileAccessConfigInstanceName := req.NodeId + "-" + req.VolumeId
+	cnsFileAccessConfigInstanceKey := types.NamespacedName{
+		Namespace: c.supervisorNamespace,
+		Name:      cnsFileAccessConfigInstanceName,
+	}
+	var newlyCreatedCnsFileAccessConfigInstance bool
+	// Check whether the CnsFileAccessConfig instance exist in the supervisor cluster
+	if err := c.cnsOperatorClient.Get(ctx, cnsFileAccessConfigInstanceKey, cnsFileAccessConfigInstance); err != nil {
+		if !errors.IsNotFound(err) {
+			// Get() on the CnsFileAccessConfig instance failed with different error
+			msg := fmt.Sprintf("failed to get CnsFileAccessConfig instance: %q/%q. Error: %+v", c.supervisorNamespace, cnsFileAccessConfigInstance.Name, err)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
+		// Create the CnsFileAccessConfig instance since it is not found
+		cnsFileAccessConfigInstance = &cnsfileaccessconfigv1alpha1.CnsFileAccessConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      cnsFileAccessConfigInstanceName,
+				Namespace: c.supervisorNamespace},
+			Spec: cnsfileaccessconfigv1alpha1.CnsFileAccessConfigSpec{
+				VMName:  req.NodeId,
+				PvcName: req.VolumeId,
+			},
+		}
+		log.Debugf("Creating CnsFileAccessConfig instance: %+v", cnsFileAccessConfigInstance)
+		if err := c.cnsOperatorClient.Create(ctx, cnsFileAccessConfigInstance); err != nil {
+			msg := fmt.Sprintf("failed to create cnsFileAccessConfig: %q/%q. Error: %v", c.supervisorNamespace, cnsFileAccessConfigInstance.Name, err)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
+		newlyCreatedCnsFileAccessConfigInstance = true
```

Comment:
Why is this variable needed? It doesnt add much value to like 498..

Reply:
Doesn't add much value. Removed